src/Data/List/Properties.agda

```diff
@@ -824,6 +824,13 @@ drop-take-suc-tabulate : ∀ {n} (f : Fin n → A) (i : Fin n) → let m = toℕ
 drop-take-suc-tabulate f i rewrite sym (toℕ-cast (sym (length-tabulate f)) i) | sym (lookup-tabulate f i)
   = drop-take-suc (tabulate f) (cast _ i)
 
+-- Dropping m elements and then n elements is same as dropping n+m elements
+drop-drop : (n m : ℕ) → (x : List A) → drop n (drop m x) ≡ drop (n + m) x
```

Comment:
The proof is probably a lot easier if you reformulate it as `drop n (drop m xs)  ≡ drop (m + n) xs`.
(Also can we please stick to plural names for lists?)

Reply:
Thanks for the review! By a lot easier, do you mean less lines or less complexity? The following variants work -

```agda
drop-fusion' : (n m : ℕ) → (xs : List A) → drop n (drop m xs) ≡ drop (m + n) xs
drop-fusion' ℕ.zero m xs rewrite +-identityʳ m = refl
drop-fusion' (suc n) ℕ.zero [] = refl
drop-fusion' (suc n) (suc m) [] = refl
drop-fusion' (suc n) ℕ.zero (x ∷ xs) = refl
drop-fusion' (suc n) (suc m) (x ∷ xs) = drop-fusion' (suc n) m xs
```

```agda
drop-fusion' : (n m : ℕ) → (xs : List A) → drop n (drop m xs) ≡ drop (m + n) xs
drop-fusion' ℕ.zero m xs rewrite +-identityʳ m = refl
drop-fusion' (suc n) ℕ.zero xs = refl
drop-fusion' (suc n) (suc m) [] = refl
drop-fusion' (suc n) (suc m) (x ∷ xs) = drop-fusion' (suc n) m xs
```

```agda
drop-fusion' : (n m : ℕ) → (xs : List A) → drop n (drop m xs) ≡ drop (m + n) xs
drop-fusion' n ℕ.zero [] = refl
drop-fusion' n (suc m) [] = drop-[] n
drop-fusion' ℕ.zero m (x ∷ xs) rewrite +-identityʳ m = refl
drop-fusion' (suc n) ℕ.zero (x ∷ xs) = refl
drop-fusion' (suc n) (suc m) (x ∷ xs) = drop-fusion' (suc n) m xs
```

Out of curiosity, how is one supposed to choose between multiple correct definitions for the same proof?

Reply:
I mean that you don't need to split on `n` because both `_+_` will compute just
enough for the inner drop to step. Cf.

```
drop-drop : (m n : ℕ) → (xs : List A) → drop n (drop m xs) ≡ drop (m + n) xs
drop-drop zero n xs = refl
drop-drop (suc m) n [] = drop-[] n
drop-drop (suc m) n (x ∷ xs) = drop-drop m n xs
```

Reply:
Ah, thanks for the clarification! I'll get this in, that was very helpful!

Reply:
Okay now I know where I went wrong - I swapped `n` and `m` _only_ in `drop (n + m) xs` and the type checker refused to simplify the proof. Spent a lot of time in confusion but everything works now! 😬